lib/index.js

```diff
@@ -39,6 +40,151 @@ Wigzo.prototype.loaded = function() {
 };
 
 /**
+ * Product Added
+ *
+ * @param {Track} track
+ */
+
+Wigzo.prototype.productAdded = function(track) {
+  var productId = track.productId();
+  if (productId) window.wigzo.track('addtocart', productId);
+};
+
+
+/**
+ * Wishlist Product Added to Cart
+ *
+ * @param {Track} track
+ */
+
+Wigzo.prototype.productAddedFromWishlistToCart = function(track) {
+  var productId = track.productId();
+  if (productId) window.wigzo.track('addtocart', productId);
+};
+
+/**
+ * Product Removed
+ *
+ * @param {Track} track
+ */
+
+Wigzo.prototype.productRemoved = function(track) {
+  var productId = track.productId();
+  if (productId) window.wigzo.track('removedfromcart', productId);
+};
+
+/**
+ * Products Searched
+ *
+ * @param {Track} track
+ */
+
+Wigzo.prototype.productsSearched = function(track) {
+  var props = track.properties();
+  if (props.query) {
+    window.wigzo.track('search', props.query);
+  }
+};
+
+/**
+ * Product Added to Wishlist
+ *
+ * @param {Track} track
+ */
+
+Wigzo.prototype.productAddedToWishlist = function(track) {
+  var productId = track.productId();
+  if (productId) window.wigzo.track('wishlist', productId);
+};
+
+
+/**
+ * Checkout Started
+ *
+ * @param {Track} track
+ */
+
+Wigzo.prototype.checkoutStarted = function(track) {
+  var productList = track.products();
+  var addedProductIds = [];
+  for (var i = 0; i < productList.length; i++) {
+    var item = productList[i];
+    addedProductIds.push(item.product_id || item.productId);
+  }
+
+  if (addedProductIds.length) window.wigzo.track('checkoutstarted', addedProductIds);
+};
+
+/**
+ * Completed Order
+ *
+ * @param {Track} track
+ */
+
+Wigzo.prototype.orderCompleted = function(track) {
+  var productList = track.products();
+  var addedProductIds = [];
+  for (var i = 0; i < productList.length; i++) {
+    var item = productList[i];
+    addedProductIds.push(item.product_id || item.productId);
+  }
+
+  if (addedProductIds.length) window.wigzo.track('buy', addedProductIds);
+};
+
+/**
+ * Product Review
+ *
+ * @param {Track} track
+ */
+Wigzo.prototype.productReviewed = function(track) {
+  window.wigzo.track('review', track.properties());
+};
+
+/**
+ * Product Clicked
+ *
+ * @param {Track} track
+ */
+Wigzo.prototype.productClicked = function(track) {
+  var props = track.properties();
+  var traits = reject({
+    productId : track.productId(),
+    title: track.name(),
+    price: track.currency() + ' ' + track.price(),
+    category: track.category(),
+    /* custom props */
+    image: props.imageUrl,
```

Comment:
so the custom props will not be inside `properties`

this is how you extract integration options:

```
var options = track.options(this.name); 
...

image: options.imageUrl
```
However, integration options are _optional_ and thus you should probably write code like this:

```
var options = track.options(this.name); 
...

if (options.imageUrl) image: options.imageUrl
```